# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursor.cpp
# Commit: f4d11e6a6e3f
# Full Hash: f4d11e6a6e3f74cd0f43c2a657b596b1eb4a1e55
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-03-26 15:34:27
# Regressor Bug: 1624810
# File Overlap Count: 1
# Description:
#   Bug 1624810 - Set the multi-zone JS holder flag on appropriate cycle collected classes r=mccr8
#   
#   Set the flag on the affected classes, which are:
#    - CallbackTimeoutHandler
#    - nsJSArgArray
# ==============================================================================

diff -r 4dcdeed2265f -r f4d11e6a6e3f dom/indexedDB/IDBCursor.cpp
--- a/dom/indexedDB/IDBCursor.cpp	Thu Mar 26 10:48:30 2020 +0000
+++ b/dom/indexedDB/IDBCursor.cpp	Thu Mar 26 10:49:06 2020 +0000
@@ -812,9 +812,7 @@
 
 // Don't unlink mRequest or mSource in
 // NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED!
-#define NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(_subclassName)            \
-  NS_IMPL_CYCLE_COLLECTION_CLASS(_subclassName)                               \
-                                                                              \
+#define NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS_METHODS(_subclassName)    \
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN_INHERITED(_subclassName, IDBCursor) \
     NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mSource)                                \
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE_END                                       \
@@ -830,9 +828,18 @@
   NS_IMPL_ADDREF_INHERITED(_subclassName, IDBCursor)                          \
   NS_IMPL_RELEASE_INHERITED(_subclassName, IDBCursor)
 
+#define NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(_subclassName) \
+  NS_IMPL_CYCLE_COLLECTION_CLASS(_subclassName)                    \
+  NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS_METHODS(_subclassName)
+
+#define NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_MULTI_ZONE_JSHOLDER_SUBCLASS( \
+    _subclassName)                                                       \
+  NS_IMPL_CYCLE_COLLECTION_MULTI_ZONE_JSHOLDER_CLASS(_subclassName)      \
+  NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS_METHODS(_subclassName)
+
 NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(IDBObjectStoreCursor)
 NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(IDBObjectStoreKeyCursor)
-NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(IDBIndexCursor)
+NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_MULTI_ZONE_JSHOLDER_SUBCLASS(IDBIndexCursor)
 NS_IMPL_CYCLE_COLLECTION_IDBCURSOR_SUBCLASS(IDBIndexKeyCursor)
 
 template <IDBCursor::Type CursorType>